### ISSUE TITLE
Filtering samplings gives complement result

### ISSUE THREAD

**Issue by schuhschuh** [openmole/openmole#74]
I just discovered the following issue. I believe this was introduced more recently.

Given the OpenMOLE script:
```scala
val i = Val[Int]
val j = Val[Int]
val si = i in (1 to 10)
val sj = j in (1 to 10)
val e = ExplorationTask((si x sj) filter "i < j")
val ex = e -< (EmptyTask() set (inputs += (i, j), outputs += (i, j)) hook ToStringHook()) start
```
This produces the following output, which would rather correspond to the filter condition "i >= j" (i.e., the negation of what I wanted. And in fact, when I use "i >= j" as filter condition, I get what I initially intended. I am sure that the sampling filter before used to keep all the elements for which the filter condition evaluated to true, not the other way around as it is right now (because I usually used "i != j" and would have only gotten a single sample, but that wasn't the case).

```
{i=10, j=10}
{i=10, j=9}
{i=10, j=8}
{i=10, j=7}
{i=10, j=6}
{i=10, j=5}
{i=10, j=4}
{i=10, j=3}
{i=10, j=2}
{i=10, j=1}
{i=9, j=9}
{i=9, j=8}
{i=9, j=7}
{i=9, j=6}
{i=9, j=5}
{i=9, j=4}
{i=9, j=3}
{i=9, j=2}
{i=9, j=1}
{i=8, j=8}
{i=8, j=7}
{i=8, j=6}
{i=8, j=5}
{i=8, j=4}
{i=8, j=3}
{i=8, j=2}
{i=8, j=1}
{i=7, j=7}
{i=7, j=6}
{i=7, j=5}
{i=7, j=4}
{i=7, j=3}
{i=7, j=2}
{i=7, j=1}
{i=6, j=6}
{i=6, j=5}
{i=6, j=4}
{i=6, j=3}
{i=6, j=2}
{i=6, j=1}
{i=5, j=5}
{i=5, j=4}
{i=5, j=3}
{i=5, j=2}
{i=5, j=1}
{i=4, j=4}
{i=4, j=3}
{i=4, j=2}
{i=4, j=1}
{i=3, j=3}
{i=3, j=2}
{i=3, j=1}
{i=2, j=2}
{i=2, j=1}
{i=1, j=1}
```

**Comment by romainreuillon** [comment]
Anyway it is better if it is consistent with scala filter semantic, it has been fixed in bf9a64f0ca118b4c58ac0db729e7546f9fc9d05c.

**Comment by schuhschuh** [comment]
Thanks!
